### PR TITLE
multi-caption

### PR DESCRIPTION
**Pull request by bghira**

### CODE REVIEW

File: helpers/image_manipulation/training_sample.py [hunk 1]
```diff
@@ -691,7 +691,7 @@ def get_conditioning_image(self):
         conditioning_dataset = StateTracker.get_conditioning_dataset(
             data_backend_id=self.data_backend_id
         )
-        print(f"Conditioning dataset components: {conditioning_dataset.keys()}")
+        # print(f"Conditioning dataset components: {conditioning_dataset.keys()}")
```
Comment (sayakpaul) [inline helpers/image_manipulation/training_sample.py:694]: Maybe we can just log this information formally?

File: helpers/metadata/backends/parquet.py [hunk 2]
```diff
@@ -136,6 +136,12 @@ def _extract_captions_to_fast_list(self):
                 filename = str(row[filename_column])
             else:
                 filename = str(index)
+            if type(caption_column) == list:
+                caption = None
+                if len(caption_column) > 0:
+                    caption = [row[c] for c in caption_column]
+            else:
+                caption = row[caption_column]
```
Comment (sayakpaul) [inline helpers/metadata/backends/parquet.py:147]: Do we wanna randomize the caption sampling part similar to [this](https://github.com/huggingface/diffusers/blob/4ad7a1f5fd48ce3aabc46f090f28b56a02e733aa/examples/text_to_image/train_text_to_image.py#L780)?

File: helpers/prompts.py [hunk 3]
```diff
@@ -255,7 +254,10 @@ def prepare_instance_prompt_from_parquet(
         if image_caption:
             image_caption = image_caption.strip()
         if prepend_instance_prompt:
-            image_caption = instance_prompt + " " + image_caption
+            if type(image_caption) == list:
+                image_caption = [instance_prompt + " " + x for x in image_caption]
+            else:
+                image_caption = instance_prompt + " " + image_caption
```
Comment (sayakpaul) [inline helpers/prompts.py:260]: Could be nice to have a proper template class that could accept style and subject keywords along with the main body of the caption. So, for example:

image

(Figure from https://arxiv.org/abs/2402.12004)

**Review by sayakpaul — COMMENTED** [review]
From the looks of it, I don't see anything obviously weird. What are we supporting here to be specific?

File: helpers/prompts.py (hunk 3 above)
Comment (bghira) [reply helpers/prompts.py:260]: i'm not sure if { } can be in a filename, can it?

File: helpers/metadata/backends/parquet.py (hunk 2 above)
Comment (bghira) [reply helpers/metadata/backends/parquet.py:147]: this job is to retrieve all captions for caching

File: helpers/image_manipulation/training_sample.py (hunk 1 above)
Comment (bghira) [reply helpers/image_manipulation/training_sample.py:694]: it wasn't needed, but it's nice to have sometimes commented out like this if i'm adding a new model or training / validation style

**Comment by bghira** [comment]
this is to support datasets with multiple captions per sample via `textfile` caption strategy which loads them from `.txt` files. and the parquet backend. but i'm waiting for it to actually be tested with a multi-caption dataset.

**Pull request merged**